src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -142,28 +142,46 @@ private void ResetSelection()
 			}
 		}
 
-		internal override void OnSelectorItemIsSelectedChanged(SelectorItem container, bool oldIsSelected, bool newIsSelected)
+		internal override void ChangeSelectedItem(object item, bool oldIsSelected, bool newIsSelected)
 		{
 			if (!_modifyingSelectionInternally)
 			{
-				var item = ItemFromContainer(container);
 
-				if (!newIsSelected)
+				//Handle selection
+				switch (SelectionMode)
 				{
-					SelectedItems.Remove(item);
-				}
-				else
-				{
-					if (!SelectedItems.Contains(item))
-					{
-						if (!IsSelectionMultiple
-							&& SelectedItems.FirstOrDefault() is object selectedItem)
+					case ListViewSelectionMode.None:
+						break;
+					case ListViewSelectionMode.Single:
+						if (!newIsSelected)
+						{
+							SelectedIndex = -1;
+						} else
 						{
-							SelectedItems.Remove(selectedItem);
+							var t = IndexFromItem(item);
```

Comment:
Codacy found an issue: [Remove the unused local variable 't'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6215121)